src/libs/actions/SharedItem.js

```diff
@@ -0,0 +1,23 @@
+import Onyx from 'react-native-onyx';
+import ONYXKEYS from '../../ONYXKEYS';
+
+/**
+ * Saves shared item to storage.
+ *
+ * @param {Object} sharedItem
+ */
+function set(sharedItem) {
+    Onyx.set(ONYXKEYS.SHARED_ITEM, sharedItem);
+}
+
+/**
+ * Clears shared item from storage.
+ */
+function clear() {
+    Onyx.set(ONYXKEYS.SHARED_ITEM, null);
```

Comment:
> Can also use Onyx.remove().
> Though I'm not sure I would have gone with Onyx here. Seems easier to just store the shared item in memory.

@marcaaron
Could you please elaborate? I'm not sure what do you mean by storing the shared item in memory?
Should it be some exported variable from ShareManager and imported to the page? Or something else?

Unfortunately, I've spent several years in a whole different world with react-navigation or other navigator (where I could use navigator arguments for passing data to the page) and redux as single source of truth for data.
I'm fighting with and sometimes losing to patterns etched on my mind.

Reply:
Sure thing. I just meant that there is perhaps no reason to store the item. Since we will get it then either use it immediately or not. So, perhaps the same could be achieved with some callback method or temporary variable rather than going through `Onyx` which by default will commit everything to device storage.

Reply:
Yeah, but I'm receiving the share event in the listener, and then I'm not using it immediately, but redirecting to the Share Page. And I need to pass it somehow from listener to the page.

What do you think about storing the item in the variable in `ShareManager.js` and exporting HOC from ShareManager, that will pass the item to the page?

```
export default withSharedItem(SharePage);
```

Reply:
Ok I see the problem now. I think maybe it's fine to just use the Onyx for this as it requires a more global state.